Server/mods/deathmatch/logic/CElementDeleter.cpp

```diff
@@ -19,6 +19,12 @@ void CElementDeleter::Delete(class CElement* pElement, bool bUnlink, bool bUpdat
     {
         if (!IsBeingDeleted(pElement))
         {
+            // Check if the element is a team and remove players
+            if (IS_TEAM(pElement))
+            {
+               ((CTeam*)pElement)->RemoveAllPlayers();
+            }
```

Comment:
Put it **after** the trigger of `onElementDestroy` to make sure we don't brake scripts that needs the list of the players at that specific timing.